lib/marqeta/transaction.rb

```diff
@@ -60,5 +80,15 @@ def accessible_time_attributes
     def card_acceptor_hash
       attributes_hash['card_acceptor']
     end
+
+    def gateway_response_code
+      attributes_hash['gpa_order']['funding']['gateway_log']['response']['code']
```

Comment:
What's the intention here?

Reply:
@andyw8 hmm, maybe I should put a comment here. Marqeta seems to be unpredictable.
Sometimes they send `gpa_order` without `funding`, sometimes `funding` without `gateway_log`, etc. Can't seem to find a pattern for this behaviour.
For our purpose, the response code should be there or it shouldn't.
That's why I return `nil` if it fails to find the code.

Reply:
Hmm, that's concerning. Does the documentation say what _should_ be there?

Reply:
@andyw8 Unfortunately, it doesn't. The way I see it is if there is no gateway log response code, we just look at the transaction response code (which is less specific)